src/handlers.js

```diff
@@ -167,38 +195,53 @@ async function statusHandler(message) {
   let debug = Debug(debugPrefix + ':' + build.eventId);
   debug(`Handling state change for task-group ${taskGroupId}`);
 
-  let state = 'success';
+  let groupState = 'success';
 
-  if (message.exchange.endsWith('task-group-resolved')) {
-    let queue = new taskcluster.Queue({
-      rootUrl: this.context.cfg.taskcluster.rootUrl,
+  let queue = new taskcluster.Queue({
+    rootUrl: this.context.cfg.taskcluster.rootUrl,
+  });
+  let params = {};
+  do {
+    let group = await queue.listTaskGroup(taskGroupId, params);
+    params.continuationToken = group.continuationToken;
+    group.tasks.forEach(task => {
+      if (_.includes(['failed', 'exception'], task.status.state)) {
+        groupState = 'failure';
+      }
     });
-    let params = {};
-    do {
-      let group = await queue.listTaskGroup(message.payload.taskGroupId, params);
-      params.continuationToken = group.continuationToken;
-      group.tasks.forEach(task => {
-        if (_.includes(['failed', 'exception'], task.status.state)) {
-          state = 'failure';
-        }
-      });
-    } while (params.continuationToken);
-  }
-
-  if (message.exchange.endsWith('task-exception') || message.exchange.endsWith('task-failed')) {
-    state = 'failure';
-  }
+  } while (params.continuationToken);
 
   await build.modify(b => {
     if (b.state !== 'failure') {
-      b.state = state;
+      b.taskState = groupState;
```

Comment:
I don't see this field renamed in `src/data.js`.

Reply:
good catch! 🐟